File: sdk/php/src/Resources/AiFactory.php
<?php

namespace HiTechCloud\SDK\Resources;

class AiFactory extends BaseResource
{
    /**
     * List templates
     */
    public function listTemplates(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/templates');
    }

    /**
     * Save template
     */
    public function saveTemplate(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/templates', $data);
    }

    /**
     * Get provisioning mode
     */
    public function getProvisioningMode(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/hitechcloud/mode');
    }

    /**
     * Get JSON examples
     */
    public function getJsonExamples(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/hitechcloud/examples');
    }

    /**
     * Get API schema
     */
    public function getApiSchema(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/hitechcloud/schema');
    }

    /**
     * List featured templates
     */
    public function listFeaturedTemplates(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/templates/featured');
    }

    /**
     * Get template
     */
    public function getTemplate(int $serviceId, string $templateId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/templates/' . $templateId);
    }

    /**
     * Update template
     */
    public function updateTemplate(int $serviceId, string $templateId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/templates/' . $templateId . '/update', $data);
    }

    /**
     * Delete template
     */
    public function deleteTemplate(int $serviceId, string $templateId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/templates/' . $templateId . '/delete');
    }

    /**
     * List instances
     */
    public function listInstances(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/instances');
    }

    /**
     * Create instance
     */
    public function createInstance(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instances', $data);
    }

    /**
     * List instance types
     */
    public function listInstanceTypes(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/instances/types');
    }

    /**
     * Get instance
     */
    public function getInstance(int $serviceId, string $instanceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/instances/' . $instanceId);
    }

    /**
     * Update instance
     */
    public function updateInstance(int $serviceId, string $instanceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instances/' . $instanceId . '/update', $data);
    }

    /**
     * Restart instance
     */
    public function restartInstance(int $serviceId, string $instanceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instances/' . $instanceId . '/restart');
    }

    /**
     * Delete instance
     */
    public function deleteInstance(int $serviceId, string $instanceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instances/' . $instanceId . '/delete');
    }

    /**
     * Get service instance
     */
    public function getServiceInstance(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/instance');
    }

    /**
     * Sync service instance
     */
    public function syncServiceInstance(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instance/sync');
    }

    /**
     * Restart service instance
     */
    public function restartServiceInstance(int $serviceId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instance/restart');
    }

    /**
     * Update service instance
     */
    public function updateServiceInstance(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/instance/update', $data);
    }

    /**
     * List SSH keys
     */
    public function listSshKeys(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/sshkeys');
    }

    /**
     * Add SSH key
     */
    public function addSshKey(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/sshkeys', $data);
    }

    /**
     * Get SSH key
     */
    public function getSshKey(int $serviceId, string $keyId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/sshkeys/' . $keyId);
    }

    /**
     * Delete SSH key
     */
    public function deleteSshKey(int $serviceId, string $keyId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/sshkeys/' . $keyId . '/delete');
    }

    /**
     * Set default SSH key
     */
    public function setDefaultSshKey(int $serviceId, string $keyId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/sshkeys/' . $keyId . '/setdefault');
    }

    /**
     * List volumes
     */
    public function listVolumes(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/volumes');
    }

    /**
     * Create volume
     */
    public function createVolume(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/volumes', $data);
    }

    /**
     * List volume types
     */
    public function listVolumeTypes(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/volumes/types');
    }

    /**
     * Get volume
     */
    public function getVolume(int $serviceId, string $volumeId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/volumes/' . $volumeId);
    }

    /**
     * Delete volume
     */
    public function deleteVolume(int $serviceId, string $volumeId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/volumes/' . $volumeId . '/delete');
    }

    /**
     * List clusters
     */
    public function listClusters(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/clusters');
    }

    /**
     * Create cluster
     */
    public function createCluster(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/clusters', $data);
    }

    /**
     * List cluster types
     */
    public function listClusterTypes(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/clusters/types');
    }

    /**
     * Get service cluster
     */
    public function getServiceCluster(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/cluster');
    }

    /**
     * Get cluster
     */
    public function getCluster(int $serviceId, string $clusterId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/clusters/' . $clusterId);
    }

    /**
     * Delete cluster
     */
    public function deleteCluster(int $serviceId, string $clusterId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/clusters/' . $clusterId . '/delete');
    }

}

File: sdk/csharp/Resources/AiFactory.cs
namespace HiTechCloud.SDK.Resources;

public class AiFactory : BaseResource
{
    public AiFactory(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> TemplatesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/templates", id));
    }

    public async Task<Dictionary<string, object?>> SaveTemplateAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/templates", id), data);
    }

    public async Task<Dictionary<string, object?>> ProvisioningModeAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/hitechcloud/mode", id));
    }

    public async Task<Dictionary<string, object?>> ExamplesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/hitechcloud/examples", id));
    }

    public async Task<Dictionary<string, object?>> SchemaAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/hitechcloud/schema", id));
    }

    public async Task<Dictionary<string, object?>> FeaturedTemplatesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/templates/featured", id));
    }

    public async Task<Dictionary<string, object?>> TemplateAsync(int id, int templateId)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/templates/{templateId}", id, templateId));
    }

    public async Task<Dictionary<string, object?>> UpdateTemplateAsync(int id, int templateId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/templates/{templateId}", id, templateId), data);
    }

    public async Task<Dictionary<string, object?>> DeleteTemplateAsync(int id, int templateId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/templates/{templateId}/delete", id, templateId), null);
    }

    public async Task<Dictionary<string, object?>> InstancesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/instances", id));
    }

    public async Task<Dictionary<string, object?>> CreateInstanceAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instances", id), data);
    }

    public async Task<Dictionary<string, object?>> InstanceTypesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/instances/types", id));
    }

    public async Task<Dictionary<string, object?>> InstanceAsync(int id, int instanceId)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/instances/{instanceId}", id, instanceId));
    }

    public async Task<Dictionary<string, object?>> UpdateInstanceAsync(int id, int instanceId, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instances/{instanceId}/update", id, instanceId), data);
    }

    public async Task<Dictionary<string, object?>> RestartInstanceAsync(int id, int instanceId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instances/{instanceId}/restart", id, instanceId), null);
    }

    public async Task<Dictionary<string, object?>> DeleteInstanceAsync(int id, int instanceId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instances/{instanceId}/delete", id, instanceId), null);
    }

    public async Task<Dictionary<string, object?>> ServiceInstanceAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/instance", id));
    }

    public async Task<Dictionary<string, object?>> SyncInstanceAsync(int id)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instance/sync", id), null);
    }

    public async Task<Dictionary<string, object?>> RestartServiceAsync(int id)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instance/restart", id), null);
    }

    public async Task<Dictionary<string, object?>> UpdateServiceAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/instance/update", id), data);
    }

    public async Task<Dictionary<string, object?>> SshKeysAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/sshkeys", id));
    }

    public async Task<Dictionary<string, object?>> CreateSshKeyAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/sshkeys", id), data);
    }

    public async Task<Dictionary<string, object?>> SshKeyAsync(int id, int keyId)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/sshkeys/{keyId}", id, keyId));
    }

    public async Task<Dictionary<string, object?>> DeleteSshKeyAsync(int id, int keyId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/sshkeys/{keyId}/delete", id, keyId), null);
    }

    public async Task<Dictionary<string, object?>> SetDefaultSshKeyAsync(int id, int keyId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/sshkeys/{keyId}/setdefault", id, keyId), null);
    }

    public async Task<Dictionary<string, object?>> VolumesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/volumes", id));
    }

    public async Task<Dictionary<string, object?>> CreateVolumeAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/volumes", id), data);
    }

    public async Task<Dictionary<string, object?>> VolumeTypesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/volumes/types", id));
    }

    public async Task<Dictionary<string, object?>> VolumeAsync(int id, int volumeId)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/volumes/{volumeId}", id, volumeId));
    }

    public async Task<Dictionary<string, object?>> DeleteVolumeAsync(int id, int volumeId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/volumes/{volumeId}/delete", id, volumeId), null);
    }

    public async Task<Dictionary<string, object?>> ClustersAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/clusters", id));
    }

    public async Task<Dictionary<string, object?>> CreateClusterAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/clusters", id), data);
    }

    public async Task<Dictionary<string, object?>> ClusterTypesAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/clusters/types", id));
    }

    public async Task<Dictionary<string, object?>> ServiceClusterAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/cluster", id));
    }

    public async Task<Dictionary<string, object?>> ClusterAsync(int id, int clusterId)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/clusters/{clusterId}", id, clusterId));
    }

    public async Task<Dictionary<string, object?>> DeleteClusterAsync(int id, int clusterId)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/clusters/{clusterId}/delete", id, clusterId), null);
    }
}
